backend/base/interfaces.go

```diff
@@ -0,0 +1,14 @@
+package base
+
+type ISearchService interface {
```

Comment:
No `ISomething`s please.
The interface in go should be defined at the place where it is used, so something like the "base" having an interface makes very little sense."

We can take a closer look at this on Monday.